pydantic_ai_slim/pydantic_ai/agent.py

```diff
@@ -96,352 +97,278 @@
 RunOutputDataT = TypeVar('RunOutputDataT')
 """Type variable for the result data of a run where `output_type` was customized on the run call."""
 
+EventStreamHandler: TypeAlias = Callable[
+    [
+        'AbstractAgent[AgentDepsT, OutputDataT]',
+        RunContext[AgentDepsT],
+        AsyncIterable[Union[_messages.AgentStreamEvent, _messages.HandleResponseEvent]],
```

Comment:
I'm inclined to merge `HandleResponseEvent` into `AgentStreamEvent` or to add a new union type, but what would it be called other than `AgentStreamEvent`...?